docs/github-actions-testing-guide.md

```diff
@@ -0,0 +1,547 @@
+# Testing GitHub Actions with `act`
+
+This guide explains how to test GitHub Actions workflows locally using [`act`](https://github.com/nektos/act), a tool that runs your workflows locally using Podman (or Docker).
+
+## What is `act`?
+
+`act` is a CLI tool that allows you to run GitHub Actions workflows locally. It uses Podman (or Docker) to simulate the GitHub Actions runner environment, making it easier to test and debug workflows before pushing to GitHub.
+
+**Note**: While GitHub Actions uses Docker, this guide uses Podman for local testing as it's rootless and better suited for local development environments.
+
+## Installation
+
+### Linux/macOS
+
+```bash
+# Using Homebrew (macOS/Linux)
+brew install act
+
+# Using curl
+curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
+
+# Using Go
+go install github.com/nektos/act@latest
+```
+
+### Verify Installation
+
+```bash
+act --version
+```
+
+## Prerequisites
+
+- **Podman** (recommended for local testing) or **Docker** (fallback)
+- **Git** (for cloning repositories)
+- **Python 3.12+** (for running tests)
+
+**Note**: GitHub Actions workflows use Docker, but for local testing with `act`, Podman is recommended as it's rootless and works better in local environments.
+
+## Quick Start
+
+### 1. Install `act`
+
+```bash
+# macOS/Linux with Homebrew
+brew install act
+
+# Or using curl
+curl https://raw.githubusercontent.com/nektos/act/master/install.sh | sudo bash
+```
+
+### 1.5. Ensure Podman is Available (Recommended)
+
+```bash
+# Check if Podman is installed
+podman --version
+
+# If not installed (Fedora/RHEL):
+sudo dnf install podman
+
+# Verify Podman is working
+podman ps
+```
+
+**Note**: The helper script (`scripts/test-with-act.sh`) will automatically detect and use Podman if available, falling back to Docker if needed. GitHub Actions workflows use Docker, but Podman is preferred for local testing.
```

Comment:
This sentence "GitHub Actions workflows use Docker, but Podman is preferred for local testing." is duplicated several times.